### PR TITLE
Bjorncs/capabilities

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: security-utils/src/test/java/com/yahoo/security/tls/ConnectionAuthContextTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package com.yahoo.security.tls;// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (vekterli) [inline security-utils/src/test/java/com/yahoo/security/tls/ConnectionAuthContextTest.java:1]: Slight mangling of `package` and copyright header

File: security-utils/src/test/java/com/yahoo/security/tls/ConnectionAuthContextTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    private static X509Certificate createCertificate() {
+        KeyPair keyPair = KeyUtils.generateKeypair(KeyAlgorithm.EC, 256);
+        return X509CertificateBuilder.fromKeypair(
+                        keyPair, new X500Principal("CN=myidentity"), Instant.EPOCH,
+                        Instant.EPOCH.plus(100000, ChronoUnit.DAYS), SHA256_WITH_ECDSA, BigInteger.ONE)
+                .build();
+    }
+
+
+}
```
Comment (vekterli) [inline security-utils/src/test/java/com/yahoo/security/tls/ConnectionAuthContextTest.java:59]: Nit: missing newline

File: security-utils/src/main/java/com/yahoo/security/tls/ConnectionAuthContext.java [hunk 3]
```diff
@@ -7,28 +7,77 @@
...
+    }
 
     public boolean authorized() { return !capabilities.hasNone(); }
 
+    public boolean hasCapabilities(CapabilitySet requiredCapabilities) {
+        return hasCapabilities(requiredCapabilities, null, null, null);
+    }
+
+    /** Provided strings are used for improved logging only */
+    public boolean hasCapabilities(CapabilitySet requiredCapabilities, String action, String resource, String peer) {
```
Comment (vekterli) [inline security-utils/src/main/java/com/yahoo/security/tls/ConnectionAuthContext.java:44]: `hasCapabilities` sounds like a very pure and innocent function with no side effects. Maybe it should be called `checkCapabilities` or something similar? Though that in turn might sound like an exception is thrown on failure... Consider factoring out the actual mode checking and logging once we see how the capability checking piece fits into the grander authz puzzle.

**Review by vekterli — APPROVED** [review]
:+1:

**Comment by bjorncs** [comment]
@vekterli PTAL

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**